ui/app/src/model/admin-client.ts

```diff
@@ -89,10 +89,14 @@ export function useGlobalDatasource(name: string) {
  * Used to get global datasources from the API.
  * Will automatically be refreshed when cache is invalidated
  */
-export function useGlobalDatasourceList() {
-  return useQuery<GlobalDatasourceResource[], Error>([globalDatasourceResource], () => {
-    return getGlobalDatasources();
-  });
+export function useGlobalDatasourceList(refetchOnMount = true) {
```

Comment:
I think it would be better to have all hooks aligned with `useProjectList`
```suggestion
type GlobalDatasourceListOptions = Omit<UseQueryOptions<GlobalDatasourceResource[], Error>, 'queryKey' | 'queryFn'>; 
export function useGlobalDatasourceList(options: GlobalDatasourceListOptions) {
```